includes/class-wc-countries.php

```diff
@@ -234,10 +237,11 @@ public function get_base_postcode() {
 	/**
 	 * Get the allowed countries for the store.
 	 *
+	 * @param bool $filter_by_allowed_countries Whether to only return results based on allowed countries. Defaults to true.
 	 * @return array
 	 */
-	public function get_allowed_countries() {
-		if ( 'all' === get_option( 'woocommerce_allowed_countries' ) ) {
+	public function get_allowed_countries( $filter_by_allowed_countries = true ) {
```

Comment:
It seems "wrong" that `get_allowed_countries()` has a parameter to determine whether you want to actually only get the allowed countries. I think you could just use `apply_filters( 'woocommerce_countries_allowed_countries', $this->countries )` directly in `load_country_states()`  instead of this roundabout way. 

The `$filter_by_x` parameter seems reasonable everywhere else you've added it.

Thoughts?